### DISCUSSION TITLE
[BUG] Pydantic BaseModel literal types display correctly in openapi.json with multiple values but not with single values

### DISCUSSION THREAD

**Discussion by imSniff** [fastapi/fastapi#11119] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import json
from typing import Literal

from fastapi import FastAPI
from fastapi.openapi.utils import get_openapi
from pydantic import BaseModel


class Dog(BaseModel):
    animal_class: Literal["mammal"]
    race: Literal["Bulldog", "Bull terrier", "Beagle"]


app = FastAPI()


@app.post("")
async def insert(dog: Dog):
    pass


openapi = get_openapi(
    title=app.title,
    version=app.version,
    openapi_version=app.openapi_version,
    description=app.description,
    routes=app.routes,
)
print(json.dumps(openapi, indent=4))
```

### Description

I found a bug in the generation of openapi.json from fastapi, with the type Literal in the Pydantic BaseModel, after generating the openapi.json the type is not included when the Literal has only one value.

Actual results in openapi.json:
```json
"Dog": {
    "properties": {
        "animal_class": {"const": "mammal", "title": "Animal Class"},
        "race": {"type": "string", "enum": ["Bulldog", "Bull terrier", "Beagle"], "title": "Race"}
    },
    "type": "object",
    "required": ["animal_class", "race"],
    "title": "Dog"
}
```
Picture of redoc documentation:

![image](https://github.com/tiangolo/fastapi/assets/28787874/bd94b789-0ba3-4ffc-a931-0344471d9c87)

Expected result:
In the generation of openapi.json it should add the field `"type": "string"` like this:
```json
"Dog": {
    "properties": {
        "animal_class": {"type": "string", "const": "mammal", "title": "Animal Class"},
        "race": {"type": "string", "enum": ["Bulldog", "Bull terrier", "Beagle"], "title": "Race"}
    },
    "type": "object",
    "required": ["animal_class", "race"],
    "title": "Dog"
}
```
The result picture of redoc documentation will be like this:

![image](https://github.com/tiangolo/fastapi/assets/28787874/412ceca2-670f-48d5-bdd3-294fa19b5ac4)

### Operating System

Linux

### FastAPI Version

0.109.2

### Pydantic Version

2.5.3

### Python Version

3.12.0

**Comment by PidgeyBE** [comment]
Any workaround for this? It's still a bug it seems, with pydantic `2.11.2` :(

**Comment by YuriiMotov** [comment, marked as answer]
This has been fixed!
Works fine in current versions (FastAPI 0.115.12, Pydantic 2.11.5):

```json
{
    "properties": {
        "animal_class": {
            "type": "string",
            "const": "mammal",
            "title": "Animal Class"
        },
        "race": {
            "type": "string",
            "enum": [
                "Bulldog",
                "Bull terrier",
                "Beagle"
            ],
            "title": "Race"
        }
    },
    "type": "object",
    "required": [
        "animal_class",
        "race"
    ],
    "title": "Dog"
}
```
